spec/player_spec.rb

```diff
@@ -44,4 +53,180 @@
+        end
+    end
+
+    context 'all strategies' do
+        it 'takes the best move' do
+            smart_computer_board = [
+                Token::X, Token::EMPTY, Token::EMPTY, 
+                Token::EMPTY, Token::O, Token::EMPTY, 
+                Token::X, Token::EMPTY, Token::EMPTY
+            ]
+            board.update(position: nil, token: nil, all_positions: smart_computer_board)
+            expect(computer_player.move(board: board, presenter: presenter)).to eq 3
+        end
+    end
+
+    context '4x4' do
+
+        before(:all) do
+            @board = Board.new(size: 4)
+        end
+
+        it 'takes winning move if possible' do
+            smart_computer_board = [
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::O,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::EMPTY, Token::O, Token::EMPTY, Token::EMPTY,
+                Token::O, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+            ]
+            board.update(position: nil, token: nil, all_positions: smart_computer_board)
+            expect(computer_player.move(board: board, presenter: presenter)).to eq 6
+        end
+
+        it 'takes blocking move if possible' do
+            smart_computer_board = [
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::X, Token::EMPTY, Token::X, Token::X,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+            ]
+            board.update(position: nil, token: nil, all_positions: smart_computer_board)
+            expect(computer_player.move(board: board, presenter: presenter)).to eq 5
+        end
+
+        it 'takes corner if possible' do
+            smart_computer_board = [
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::X,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+            ]
+            board.update(position: nil, token: nil, all_positions: smart_computer_board)
+            expect(computer_player.move(board: board, presenter: presenter)).to eq(0).or eq(3).or eq(12).or eq(15)
+        end
+
+
+        it 'takes next best move to win if center or corner is not available' do
+            smart_computer_board = [
+                Token::O, Token::EMPTY, Token::EMPTY, Token::O,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::EMPTY, Token::EMPTY, Token::EMPTY, Token::EMPTY,
+                Token::X, Token::EMPTY, Token::EMPTY, Token::X,
+            ]
+            board.update(position: nil, token: nil, all_positions: smart_computer_board)
+            expect(computer_player.move(board: board, presenter: presenter)).to eq(1).or eq(2)
+        end
+    end
```

Comment:
Hey, cool, testing 4x4!  How long do these tests take to run?  Just out of curiosity

Reply:
It says the following -- just curious, is this fast or slow for a simple game like this?
```
Finished in 0.01862 seconds (files took 0.14409 seconds to load)
88 examples, 0 failures
```

Reply:
I'd say that's excellent!  💯